pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -411,27 +414,26 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 				listenerMapKey = fmt.Sprintf("%s:%d", listenAddress, servicePort.Port)
 				var exists bool
 				if currentListener, exists = listenerMap[listenerMapKey]; exists {
-					// Check if this is HTTPS port collision for external service. If so, we can use SNI to differentiate
-					// Internal TCP services will never hit this issue because they are bound by specific IP_port, while
-					// external service listeners are typically bound to 0.0.0.0
-					if !listenerTypeMap[listenerMapKey].IsTCP() || !servicePort.Protocol.IsTLS() || !service.MeshExternal {
+					// Check for port collisions between TCP/TLS and other port types.
+					// If configured correctly, TCP/TLS ports may not collide.
+					// We'll need to do additional work to find out if there is a collision for TCP/TLS.
```

Comment:
add a comment here stating that this is to check if TCP and HTTP ports are colliding..